checker/cost.go

```diff
@@ -476,6 +476,21 @@ func (c *coster) sizeEstimate(t AstNode) SizeEstimate {
 	if l := c.estimator.EstimateSize(t); l != nil {
 		return *l
 	}
+	// return a constant size if we're dealing with a CallExpr and EstimateSize
+	// came up empty; this code is reached when a CallExpr is nested inside
+	// another CallExpr and doesn't have an estimator that can handle it, such
+	// as `(1 == 2) == (3 == 4)` - without this block, that expression would
+	// return SizeEstimate{Min: 0, Max: math.MaxUint64}
+	if _, ok := t.Expr().ExprKind.(*exprpb.Expr_CallExpr); ok {
+		// ensure we only return an estimate of 1 for return types of set
+		// lengths, since strings/bytes/more complex objects could be of
+		// variable length
+		if kindOf(t.Type()) == kindPrimitive {
+			if t.Type().GetPrimitive() != exprpb.Type_STRING && t.Type().GetPrimitive() != exprpb.Type_BYTES {
```

Comment:
suggestion only: put this into a small utility function?

Reply:
Makes sense - I've refactored that code into `isScalar`.